tokio-trace/tokio-trace-core/src/dispatcher.rs

```diff
@@ -48,6 +48,7 @@ pub fn with_default<T>(dispatcher: Dispatch, f: impl FnOnce() -> T) -> T {
         }
     }
 
+    let dispatcher = dispatcher.clone();
```

Comment:
Looking at the other `with_default`s in Tokio, it seems a reference is fine (they don't clone).

Reply:
It takes some `unsafe` code, but yes. It should be safe to set the thread-local to a ref.

Reply:
we can also punt initially...

Reply:
I'm happy to change this later (as it's an implementation detail) but personally, I'd prefer not to use unsafe code here unless benchmarking actually indicates that it has a meaningful performance impact.

Reply:
@seanmonstar I should add that `tokio-timer`'s `with_default` function also does  basically clone an `Arc`: https://github.com/tokio-rs/tokio/blob/80162306e71c8561873a9c9496d65f2c1387d119/tokio-timer/src/timer/handle.rs#L98

(`HandlePriv` is actually a `Weak`, but I don't think that would work in this case)